src/components/postgresConnect.ts

```diff
@@ -499,6 +499,38 @@ class PostgresDB {
       throw err
     }
   }
+
+  public getUserPermissionNames = async (username: string) => {
+    const text = `
+    select permission_name.id, permission_name.name 
+    from "user"
+    join permission_join on "user".id = permission_join.user_id
+    join permission_name on permission_name.id = permission_join.permission_name_id
+    where "username" = $1
+    `
+    try {
+      const result = await this.query({ text, values: [username] })
+      return result.rows
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  }
+
+  public joinPermissionNameToUser = async (username: string, permissionName: string) => {
+    const text = `
+    insert into permission_join (user_id, permission_name_id) 
+    values (
+        (select id from "user" where username = $1), 
+        (select id from permission_name where name = $2))
```

Comment:
``` suggestion
        (select id from permission_name where name = $2))
        RETURNING id
```
Useful if subsequent plugins wish to interact with this record (see output additions below).

Reply:
While we are in development phase, I don't really see the need to over optimise plugins. What would be a use case for action needing permission join id ?

Reply:
In case subsequent Actions needed to do something with the new permission_join record. But sure, it's not needed for now.